### PR TITLE
Update App.pm to support $XDG_CONFIG_HOME environnement variable

### PR DESCRIPTION
**Pull request by eerielili**
Not a complete implementation of the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) but nice to have
Latest commit can close https://github.com/dave-theunsub/clamtk/issues/93

### CODE REVIEW

**Comment by eerielili** [comment]
Latest commit can close #93

**Comment by dave-theunsub** [comment]
Hi @eerielili ,

That looks good.  I'll look more at it this weekend... will be rough to do before then.

Thank you!

respectfully,  
Dave M

**Comment by dave-theunsub** [comment]
Hi @eerielili ,

There is one issue with this.  By default, this will create a "/home/user/.config/clamtk" directory.  That's okay and even desirable... but someone who has been using this program for any length of time has everything from logs, definitions, quarantine, to preferences in "/home/user/clamtk".  I don't have a nice "click here to move everything from your old directory" function.  I mean, there could be a "copy and paste this rsync command to move everything on your own" but that's not necessarily a great idea either.

respectfully,
Dave M

**Comment by eerielili** [comment]
@dave-theunsub Pushed a new commit, it should first check if there's an already existing `$HOME/.clamtk` folder for the current user

**Review by sadtab — APPROVED** [review]

**Comment by Neustradamus** [comment]
@dave-theunsub: Have you looked changes since your latest comment?

**Comment by dave-theunsub** [comment]
Sorry, I got distracted.  I'll go back to it now.

File: lib/App.pm [hunk 1]
```diff
@@ -45,6 +45,11 @@ sub get_path {
 
     # Default personal clamtk directory
     $path->{ clamtk } = $path->{ directory } . '/.clamtk';
```
Comment (ShamrockLee) [inline lib/App.pm:47]: Why do we still keep this line of assignment while re-assign the value in the next line?

Comment (dave-theunsub) [reply lib/App.pm:47]: Yeah, that looks like a mistake. That'll be in 6.18 which can probably be pushed Friday.

**Pull request merged**